### DISCUSSION TITLE
deploy on shared PHP host / use SQLPage applications offline

### DISCUSSION THREAD

**Discussion by flosigime** [sqlpage/SQLPage#21] [Q&A]
Hi, 

Thanks for this really useful app. We are planing to use it for archeological survey on field. 

We would like to know if it's possible to use it on a standart apache web server, as we have not explored yet the use of VPS. 
Do you think it could be used on an android device ? 

Thank you for your answer. 

Florent

**Comment by lovasoa** [comment]
Hi !
  I'd be happy to help you setting up your server !
  A few questions: 

 - do you already have a web server with apache installed ? If so, you can put sqlpage behind apache, but you can also run it independently from apache, just by executing it directly on the server
 - by android, do you mean accessing the web application from android ? This is definitely possible, and all the UI components are optimized for both mobile and desktop browsers

**Comment by flosigime** [comment]
Great ! 

- I only have a small webspace on a shared server where i can put traditionnal files (php, css and html). I have no access to the apache configuration. But i can try on my local computer to understand how it works. If i execute it on an apache web server, what file does i need to download ? And where does i need to put them ?
- Yes, the idea is to use it on an android. If our database is on the web, we can simply use our android terminal to access the web site, we saw the ui is responsive, and it will be perfect. But we sometimes don't have web access, so we can need a local access.  

We tested the app this afternoon, but we encountered some difficulties, as our database is not Sqlite, but Spatialite. Our major tables have a geometry column that we don't need on field but back to offce, There are some geometryConstraints that the app doesn't understand, especially in case of INSERT. 

Thank you for your help

**Comment by lovasoa** [comment]
- If you cannot run custom software on the server (if you don't have ssh access) and cannot access Apache's configuration, it will be difficult to run something other than php there. But I can help you set it up on a VPS, or even host it for you if you want.

 - SQLPage does not send the entire database to your phone when you load the page. SQLPage runs on the server, and when you are offline, you cannot access it. 

 - What is the problem exactly with geometryConstraints ? SQLPage shouldn't treat your spatialite database any differently. Did you load the spatialite extension with `SELECT load_extension('spatialite_dynamic_library_name');` ?


I see you are based in France; so am I. Do you want to schedule a call sometime this week ? Feel free to get in touch on contact at ophir dev. I'll try to help you through the configuration.

**Comment by flosigime** [comment]
- Yes, that's what i think i was. I'll look at VPS, or maybe use an apache on my local machine. 
- yes, that's what i understood. 
- no, we didn't load the spatialite extension. I don't know how to do it. 
- i found your web site. I send you a mail...
